test/test_fx_experimental.py

```diff
@@ -0,0 +1,60 @@
+import torch
+
+from torch.fx.symbolic_trace import symbolic_trace
+from torch.fx.experimental import GraphManipulation
+from torch.fx.experimental.Partitioner import Partitioner, DAG
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.jit_utils import JitTestCase
+
+
+class TestFX(JitTestCase):
+    def test_size_based_partition(self):
+        class testModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear_mod_0 = torch.nn.Linear(3, 4)
+                self.linear_mod_1 = torch.nn.Linear(4, 4)
+            def forward(self, a, b, c):
+                d = self.linear_mod_0(a)
+                e = self.linear_mod_1(b)
+                f = torch.rand(4)
+                g = torch.rand(5)
+                return c + d + e, f, g
+
+        m = testModule()
+        traced = symbolic_trace(m)
+        GraphManipulation.get_size_of_all_nodes(
+            traced,
+            [torch.rand(3), torch.rand(4), torch.rand(4)]
+        )
+        partitioner = Partitioner()
+        devices = [
+            {'name': 'dev_0', 'available_mem': 1000},
```

Comment:
As mentioned on the previous PR, consider using [NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple) to represent device. This will make it easier to construct and understand what a `device` means.

Additionally, it's not clear to me what unit `available_mem` is supposed to be? bytes? consider a name like `available_mem_bytes` or similar